security/danectl/Portfile

```diff
@@ -0,0 +1,87 @@
+                 It means securely letting the world know in advance what \
+                 your public encryption keys are by publishing them as \
+                 DNS records (TLSA, SSHFP, OPENPGPKEY, SMIMEA) in your \
+                 DNSSEC-enabled internet domain zone. This is the simplest \
+                 and most secure way to let the world know what keys to \
+                 expect when connecting to your servers. This can apply to \
+                 TLS keys, SSH host keys, and OpenPGP and S/MIME keys. \
+                 This makes it possible to prevent impersonation or \
+                 man-in-the-middle attacks. It's mostly used with mail \
+                 servers. Eventually, it could render certificate \
+                 authorities unnecessary. DNSSEC has become very easy.
+depends_lib      port:libidn2
+default_variants +tlsa +sshfp +openpgpkey +smimea
+
+variant tlsa description {Add TLSA support for TLS servers} {
+    depends_lib-append \
+        port:certbot \
+        port:openssl \
+        port:coreutils
+}
+
+variant sshfp description {Add SSHFP support for ssh servers} {
+    depends_lib-append \
+        port:openssh
+}
+
+variant openpgpkey description {Add OPENPGPKEY support for GnuPG keys} {
+    depends_lib-append \
+        port:gnupg2
+}
+
+variant smimea description {Add SMIMEA support for S/MIME keys} {
+    # Suppress a port lint warning about port:openssl appearing twice
+    if {![variant_isset tlsa]} {
+        depends_lib-append \
+            port:openssl
+    }
+}
+
+homepage         https://raf.org/${name}/
+master_sites     ${homepage}download/ \
+                 https://web.archive.org/web/20230611042305/${homepage}download/ \
+                 https://github.com/raforg/${name}/releases/download/v${version}/ \
+                 https://codeberg.org/raforg/${name}/releases/download/v${version}/
+checksums        rmd160 d597aca2b6c82b35a811830adf2c18200aa26729 \
+                 sha256 28417b939c0b59c1e069258110b9267b44ed6bea41771e6609545eca696301ae \
+                 size 54590
+
+post-patch {
+    # Set the installation prefix
+    # Suppress a reinplace runtime warning when ${prefix} is /usr/local (unlikely)
+    if {![string match /usr/local ${prefix}]} {
+        reinplace -W ${worksrcpath} "s|/usr/local|${prefix}|g" Makefile
+    }
+    # Certbot is definitely a macports port (tlsa variant dependency)
+    reinplace -W ${worksrcpath} "s|/etc/letsencrypt|${prefix}/etc/letsencrypt|g" danectl danectl.1
+    # Assume apache2 or nginx would also be macports ports, if present
+    reinplace -W ${worksrcpath} "s|/etc/apache2|${prefix}/etc/apache2|g" danectl danectl.1
+    reinplace -W ${worksrcpath} "s|/etc/nginx|${prefix}/etc/nginx|g" danectl danectl.1
+    # Don't require users to put ${prefix}/libexec/gnubin in their $PATH
+    reinplace -W ${worksrcpath} "s|sha256sum|gsha256sum|g" danectl danectl.1
+}
+
+use_configure       no
+build {}
```

Comment:
Does this port only install scripts and other files that are not compiled for a specific processor? If so, indicate that with:
```
supported_archs noarch
```

Assuming that is the case, do any of the installed files vary based on the OS version? If not, indicate that with:

```
platforms {darwin any}
```

Reply:
Thanks. That's correct. It's a shell script. It handles various operating systems without the files themselves changing. I've added the supported_archs and platforms directives.

Reply:
Thanks. That's correct. It's a shell script. It handles various operating systems without the files themselves changing. I've added the supported_archs and platforms directives.